packages/manager/src/hooks/useGravatar.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from 'react';
+
+import { checkForGravatar, getGravatarUrl } from 'src/utilities/gravatar';
+
+// set a cache to prevent duplicate requests
+const gravatarCache: Record<string, boolean> = {};
+
+/**
+ * useGravatar
+ *
+ * @description
+ * This hook checks if a user has a Gravatar associated with their email.
+ * It uses a cache to prevent duplicate requests.
+ *
+ * @param email - The email address to check for a Gravatar.
+ * @returns - A boolean indicating whether the user has a Gravatar.
+ */
+export const useGravatar = (email: string | undefined) => {
+  const [hasGravatar, setHasGravatar] = useState(false);
```

Comment:
I think I see some flickering on http://localhost:3000/profile/display

Could get the default value from the cache to minimize flickering?

For example:
```suggestion
  const [hasGravatar, setHasGravatar] = useState(
    email && gravatarCache.has(email) ? gravatarCache.get(email) : false
  );
```

Reply:
This helps with the profile page a little, but for the notification center and the events page we need to handle first load. We discussed this with @mjac0bs last week and agreed and refactoring things a little to render a placeholder  when `hasGravatar` is `undefined`

```
hasGravatar = undefined => placeholder
hasGravatar = false => icon
hasGravatar = true => gravatar
```

we also agreed we should have **one** component handling the logic above so we don't have to repeat it through the codebase. Heck, the hook could even return the JSX

Reply:
Thanks for the suggestion, @bnussman-akamai , and for summarizing, @abailly-akamai . 

[665915c](https://github.com/linode/manager/pull/10859/commits/665915c281054acef293de110b6c902aaf68fc4c) and [130aec7](https://github.com/linode/manager/pull/10859/commits/130aec79bf314adfb14209ad78d34452b811dd6c) address  the above with a `GravatarOrAvatar` component for handing the logic of what to render. In addition to width and height, the component also needed to take in either the Avatar/Gravatar components or their props to be able to display the right color/initial/size of the Avatar or Gravatar. (We also have the two Gravatar components - GravatarByUsername and GravatarByEmail, used depending on the situation.)

I have added a placeholder and some fade-ins. What I was seeing happen was that the request in `useGravatar` would return a response (so `isLoadingGravatar` was false, `hasGravatar` was true), but we were still waiting on the request to get the avatar in `GravatarByUsername` or `GravatarByEmail`, at which point the old Avatar with the UserIcon would still flicker briefly. Adding a fade in to the Gravatar components only seemed to hide that... but in some cases (Event Landing), it's a pretty long/noticable fade in. @abailly-akamai, what are your thoughts? 

Note: On the event landing page, I am seeing a bit of flickering with the default Avatar (i.e. for an Akamai system event) when logged in a user who had Gravatar enabled. Looking at that and the unit test failure now.

Reply:
This suggestion would still also help I think